### PR TITLE
Reworked hijack example section

### LINKED ISSUE #2: Ordering of hijack example
I find the ordering of section 3 hard to follow, and distracts attention from its purpose - which is to draw attention to the additional attack vectors enabled by "loose" ROAs.
I suggest that we reorder as follows:
- description of the sub-prefix attack with "loose" ROA
- specification of revised "strict" ROA
- explanation of why sub-prefix attack is mitigated
- explanation of why prefix attack is still possible but less likely to attract traffic

Happy to do a PR if agreed on principal.

### PR DESCRIPTION
**Pull request by benmaddison**
resolves #2

Total re-work of the logical ordering of the section as proposed in #2.

Whilst the laguage is largely retained, the ordering is substantially 
altered. Thus the following may be easier to review than the PR diff:

```
3.  Forged-Origin Subprefix Hijack

   A detailed description and discussion of forged-origin subprefix
   hijacks are presented here, especially considering the case when the
   subprefix is not announced in BGP.  The forged-origin subprefix
   hijack is relevant to a scenario in which:

      (1) the RPKI [RFC6480] is deployed, and

      (2) routers use RPKI origin validation to drop invalid routes
      [RFC6811], but

      (3) BGPsec [RFC8205] (or any similar method to validate the
      truthfulness of the BGP AS_PATH attribute) is not deployed.

   Note that this set of assumptions accurately describes a substantial,
   and growing, number of large Internet networks at the time writing.

   The forged-origin subprefix hijack [RFC7115] [GCHSS] is described
   here using a running example.

   Consider the IP prefix 192.168.0.0/16 which is allocated to an
   organization that also operates AS 64496.  In BGP, AS 64496
   originates the IP prefix 192.168.0.0/16 as well as its subprefix
   192.168.225.0/24.  Therefore, the RPKI should contain a ROA
   authorizing AS 64496 to originate these two IP prefixes.

   Suppose, however, the organization issues and publishes a ROA
   including a maxLength value of 24:

      ROA:(192.168.0.0/16-24, AS 64496)

   We refer to the above as a "loose ROA" since it authorizes AS 64496
   to originate any subprefix of 192.168.0.0/16 up to and including
   length /24, rather than only those prefixes that are intended to be
   announced in BGP.

   Because AS 64496 only originates two prefixes in BGP: 192.168.0.0/16
   and 192.168.255.0/24, all other prefixes authorized by the "loose
   ROA" (for instance, 192.168.0.0/24), are vulnerable to the following
   forged-origin subprefix hijack [RFC7115] [GCHSS]:

      The hijacker AS 64511 sends a BGP announcement "192.168.0.0/24: AS
      64511, AS 64496", falsely claiming that AS 64511 is a neighbor of
      AS 64496 and falsely claiming that AS 64496 originates the IP
      prefix 192.168.0.0/24.  In fact, the IP prefix 192.168.0.0/24 is
      not originated by AS 64496.

      The hijacker's BGP announcement is valid according to the RPKI,
      since the ROA (192.168.0.0/16-24, AS 64496) authorizes AS 64496 to
      originate BGP routes for 192.168.0.0/24.

      Because AS 64496 does not actually originate a route for
      192.168.0.0/24, the hijacker's route is the *only* route to the
      192.168.0.0/24.  Longest-prefix-match routing ensures that the
      hijacker's route to the subprefix 192.168.0.0/24 is always
      preferred over the legitimate route to 192.168.0.0/16 originated
      by AS 64496.

   Thus, the hijacker's route propagates through the Internet, the
   traffic destined for IP addresses in 192.168.0.0/24 will be delivered
   to the hijacker.

   The forged-origin *subprefix* hijack would have failed if a "minimal
   ROA" described below was used instead of the "loose ROA".  In this
   example, a "minimal ROA" would be:

      ROA:(192.168.0.0/16, 192.168.225.0/24, AS 64496)

   This ROA is "minimal" because it includes only those IP prefixes that
   AS 64496 originates in BGP, but no other IP prefixes [RFC6907].

   The "minimal ROA" renders AS 64511's BGP announcement invalid,
   because:

      (1) this ROA "covers" the attacker's announcement (since
      192.168.0.0/24 is a subprefix of 192.168.0.0/16), and

      (2) there is no ROA "matching" the attacker's announcement (there
      is no ROA for AS 64511 and IP prefix 192.168.0.0/24) [RFC6811].

   If routers ignore invalid BGP announcements, the minimal ROA above
   ensures that the subprefix hijack will fail.

   Thus, if a "minimal ROA" had been used, the attacker would be forced
   to launch a forged-origin *prefix* hijack in order to attract
   traffic, as follows:

      The hijacker AS 64511 sends a BGP announcement "192.168.0.0/16: AS
      64511, AS 64496", falsely claiming that AS 64511 is a neighbor of
      AS 64496.

   This forged-origin *prefix* hijack is significantly less damaging
   than the forged-origin *subprefix* hijack:

      AS 64496 legitimately originates 192.168.0.0/16 in BGP, so the
      hijacker AS 64511 is not presenting the *only* route to
      192.168.0.0/16.

      Moreover, the path originated by AS 64511 is one hop longer than
      the path originated by the legitimate origin AS 64496.

   As discussed in [LSG16], this means that the hijacker will attract
   less traffic than he would have in the forged-origin *subprefix*
   hijack, where the hijacker presents the *only* route to the hijacked
   subprefix.

   In summary, a forged-origin subprefix hijack has the same impact as a
   regular subprefix hijack, despite the increased AS_PATH length of the
   illegitimate route.  A forged-origin *subprefix* hijack is also more
   damaging than forged-origin *prefix* hijack.

```

### CODE REVIEW

**Comment by ksriram25** [comment]
Ben,

Thanks for all the hard work. 
Sorry I could not make time due to other commitments as I mentioned before.
I will look to devote some time to this in the next two weeks.

Sriram

________________________________________
